app/code/core/Mage/Sales/Model/Order.php

```diff
@@ -671,6 +671,9 @@ public function canCancel()
      */
     public function canVoidPayment()
     {
+        if ($this->getPayment() === false) {
```

Comment:
does this need to be this strict?

Reply:
In my opinion, yes. If you remove the database table row in `sales_flat_order_payment` for an order or truncate the table completely (like e.g. in dev dumps), `$payment` is false. I always check for type equality when possible. If you would like it different, please let me know. But I'm pretty sure about this change here.